src/hotspot/cpu/x86/templateTable_x86.cpp

```diff
@@ -3936,10 +3936,15 @@ void TemplateTable::_new() {
   __ load_resolved_klass_at_index(rcx, rcx, rdx);
   __ push(rcx);  // save the contexts of klass for initializing the header
 
-  // make sure klass is initialized & doesn't have finalizer
-  // make sure klass is fully initialized
+  // make sure klass is initialized
+  // init_state needs acquire, but x86 is TSO, and so we are already good.
+#ifdef _LP64
+  assert(VM_Version::supports_fast_class_init_checks(), "must support fast class initialization checks");
+  __ clinit_barrier(rcx, r15_thread, nullptr /*L_fast_path*/, &slow_case);
+#else
```

Comment:
This seems to be unrelated to this backport - this change was added for https://bugs.openjdk.org/browse/JDK-8320276